### PR TITLE
fix room preview save UX and aspect ratio

### PR DESCRIPTION
**Pull request by denisdimitrov**
- Lock image aspect ratio when resizing.
- Persist the room size when dragging
- Show indicator for saving the rooms
- Disabled rooms when saving them.

Fixes: https://github.com/sparkletown/internal-sparkle-issues/issues/371

### CODE REVIEW

File: src/components/atoms/Button/Button.styles.ts
Comment (0xdevalias) [inline src/components/atoms/Button/Button.styles.ts:14]: [NB] The sooner we can remove `styled-components` the happier I will be

File: src/pages/Account/Venue/VenueMapEdition/DraggableSubvenue.tsx
Comment (0xdevalias) [inline src/pages/Account/Venue/VenueMapEdition/DraggableSubvenue.tsx:79]: What are the changes to the `ref` here intended to do?

File: src/pages/Admin/MapPreview/MapPreview.tsx
Comment (0xdevalias) [inline src/pages/Admin/MapPreview/MapPreview.tsx:177]: [OOS] Huh.. neat that this already existed!

That means that it would be fairly trivial (in the future) to implement a concept of being able to change the aspect ratio when holding a 'modifier key', as described in https://github.com/sparkletown/internal-sparkle-issues/issues/371#issuecomment-778673339

File: src/pages/Admin/MapPreview/MapPreview.tsx
Comment (0xdevalias) [inline src/pages/Admin/MapPreview/MapPreview.tsx:183]: Perhaps instead of `isLoading` this could be called `isSaving` or similar? Would make the code easier to follow.

The nested ternary here is.. interesting. I don't necessarily hate it.. but I feel like I should for some reason.

**Review by 0xdevalias — COMMENTED** [review]
Most of my comments are NB/etc, there are a couple of questions that depending on the answer may warrant changes.

While this is more or less ready for an approve, I will hold off on giving it for now to see what the answers are.

File: src/pages/Admin/MapPreview/MapPreview.tsx
Comment (0xdevalias) [reply src/pages/Admin/MapPreview/MapPreview.tsx:177]: (Might be worth creating a new issue to backlog this part/suggestion?)

File: src/pages/Admin/MapPreview/MapPreview.tsx
Comment (denisdimitrov) [reply src/pages/Admin/MapPreview/MapPreview.tsx:183]: Yeah, I feel the same way about the nesting here. I'll figure something out to make it more readable.

File: src/pages/Account/Venue/VenueMapEdition/DraggableSubvenue.tsx
Comment (denisdimitrov) [reply src/pages/Account/Venue/VenueMapEdition/DraggableSubvenue.tsx:79]: When the room positions are being saved, they ref will die and the drag and drop will be disabled. I looked in the library on how to disabled the drag with a prop or ref but looks like it can't be disabled.

File: src/pages/Admin/MapPreview/MapPreview.tsx
Comment (denisdimitrov) [reply src/pages/Admin/MapPreview/MapPreview.tsx:177]: Backlogged in https://github.com/sparkletown/internal-sparkle-issues/issues/392

File: src/pages/Account/Venue/VenueMapEdition/Container.tsx
Comment (0xdevalias) [inline src/pages/Account/Venue/VenueMapEdition/Container.tsx:85]: What does this function add that `setDragBoxId` doesn't already do by itself? Seems like a layer of indirection that adds no value?

File: src/pages/Admin/MapPreview/MapPreview.tsx
Comment (0xdevalias) [inline src/pages/Admin/MapPreview/MapPreview.tsx:186]: [NB] This could stay as a ternary now that you've separated the `editButtonText` part out of it.

**Review by 0xdevalias — APPROVED** [review]
[CRO] A couple of minor comments, probably worth cleaning up the function one, the ternary one isn't as important.

But from there, this seems good to go (assuming it's been tested/clicked through/etc)

File: src/pages/Account/Venue/VenueMapEdition/Container.tsx
Comment (denisdimitrov) [reply src/pages/Account/Venue/VenueMapEdition/Container.tsx:85]: It's using `useCallback` that's it.

Comment (0xdevalias) [reply src/pages/Account/Venue/VenueMapEdition/Container.tsx:85]: `setDragBoxId` comes from `useState`, which is already a stable function reference. The only time wrapping it with `useCallback` makes sense is when we're 'closing around the state' to make a 'no arguments' function. But since we still have the `id` arg in the `useCallback` that is just passed through, there's no benefit to having this here.

We can remove this chunk of code and just call/pass `setDragBoxId` to the appropriate places.

**Pull request merged**
